coffea/processor/executor.py

```diff
@@ -595,37 +668,46 @@ def work_queue_executor(items, function, accumulator, **kwargs):
 
         infile_function = os.path.join(tmpdir, "function.p")
 
-        # Nothing to do?  Just return the accumulator.
-        if len(items) == 0:
-            return accumulator
-
         # Keep track of total tasks in each state.
         tasks_submitted = 0
         tasks_done = 0
-        tasks_total = len(items)
+        items_submitted = 0
+        items_done = 0
 
         print("Listening for work queue workers on port {}...".format(_wq_queue.port))
 
         # Create a dual progress bar to show submission and completion.
         submit_bar = tqdm(
-            total=tasks_total,
+            total=items_total,
             position=0,
             disable=not status,
             unit=unit,
             desc="Submitted",
         )
         complete_bar = tqdm(
-            total=tasks_total, position=1, disable=not status, unit=unit, desc=desc
+            total=items_total, position=1, disable=not status, unit=unit, desc=desc
         )
 
-        itemiter = iter(items)
+        # triplets of num of events, wall_time, memory
+        task_reports = []
 
-        # Main loop of executor
-        while tasks_done < tasks_total:
+        # ensure items looks like a generator
+        if isinstance(items, list):
+            items = iter(items)
 
-            # Submit tasks into the queue, but no more than 100 idle tasks
-            while tasks_submitted < tasks_total and _wq_queue.stats.tasks_waiting < 100:
-                item = next(itemiter)
+        # Main loop of executor
+        while items_done < items_total:
+            while items_submitted < items_total and _wq_queue.hungry():
+                if dynamic_chunksize:
+                    next(items)
```

Comment:
I believe this `next(items)` is extraneous due to the earlier comment about extraneous `yield`